ec/provider_config.go

```diff
@@ -33,22 +35,36 @@ const (
 )
 
 // configureAPI implements schema.ConfigureFunc
-func configureAPI(d *schema.ResourceData) (interface{}, error) {
+func configureAPI(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
 	timeout, err := time.ParseDuration(d.Get("timeout").(string))
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
 	if err != nil {
-		return nil, err
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to get EC client",
+			Detail:   "Invalid timeout value for EC client",
+		})
+		return nil, diags
```

Comment:
It's a good idea to return the errors that we are getting from other functions as they are. This way, what went wrong is clearer. Let's use the `FromErr` function instead :) 
```suggestion
		return nil, diag.FromErr(err)
```